### DISCUSSION TITLE
Getting an svg drawing from segment

### DISCUSSION THREAD

**Discussion by DvdMng** [ultralytics/ultralytics#5878] [Q&A]
Hello, I'm new to this framework and I would like to know how I can implement a solution in python which uses YOLOv8 segment to isolate only people in the frame and render an svg drawing which includes all the people's outline/silhouette, black on white. Thanks!

**Comment by pderrenger** [comment]
@DvdMng hello! To achieve your goal of isolating people and rendering their outlines as SVG drawings using YOLOv8, you'll need to follow these steps:

1. **Segmentation**: Use YOLOv8's segmentation capabilities to segment people in the frame.
2. **Post-processing**: Convert the segmentation masks to contours.
3. **SVG Conversion**: Convert the contours to SVG format.

Here's a concise example of how you might implement this in Python:

```python
from ultralytics import YOLO
import cv2
import numpy as np
from skimage import measure
import svgwrite

# Load a pretrained segmentation model
model = YOLO('yolov8n-seg.pt')

# Perform segmentation on an image
results = model.predict(source='path/to/image.jpg')

# Extract masks for the 'person' class (assuming 'person' is class 0)
masks = results.masks.data
person_masks = masks[results.masks.cls == 0]

# Initialize SVG drawing
dwg = svgwrite.Drawing('people.svg', profile='tiny')

# Iterate over each person's mask
for mask in person_masks:
    # Find contours of the mask
    contours = measure.find_contours(mask, 0.5)
    
    # Draw each contour as a polygon in the SVG
    for contour in contours:
        # Scale contour for image size if necessary
        contour = contour * scale_factor  # Define scale_factor based on your needs
        # Convert contour to SVG polygon points
        points = [(point[1], point[0]) for point in contour]  # Swap x and y for SVG format
        dwg.add(dwg.polygon(points, fill='black', stroke='black'))

# Save the SVG file
dwg.save()
```

In this example, `yolov8n-seg.pt` is a placeholder for the actual path to your YOLOv8 segmentation model. Make sure to replace `'path/to/image.jpg'` with the path to your image and adjust the class index if 'person' is not class 0 in your model.

Please note that this is a simplified example. You may need to adjust the code to fit your specific use case, such as handling multiple classes, scaling the contours to match the original image size, or refining the SVG output. The `skimage` and `svgwrite` packages are used here for contour finding and SVG writing, respectively. You might need to install these packages if you haven't already.

Good luck with your project! 🚀
